app/services/identity_linker.rb

```diff
@@ -1,10 +1,13 @@
 IdentityLinker = Struct.new(:user, :provider) do
   attr_reader :identity
 
-  def link_identity(nonce: nil, session_uuid: nil, ial: nil)
+  def link_identity(session_uuid: nil, **extra_attrs)
     find_or_create_identity
 
-    identity.update!(identity_attributes(nonce: nonce, session_uuid: session_uuid, ial: ial))
+    attributes = identity_attributes(session_uuid: session_uuid).
```

Comment:
sure -- thinking that this method is doing several things, so if we put all of the actions into separate methods, that might improve readability. Here is what I was thinking:

```
def link_identity(session_uuid: nil, **extra_attrs)
  attributes = merged_attributes(session_uuid, extra_attrs)
   identity.update!(attributes)
end

private

def identity
   Identity.find_or_create_by(
      service_provider: provider,
      user_id: user.id
    )
end

def merged_attributes(session_uuid, extra_attrs)
  identity_attributes(session_uuid: session_uuid).merge(optional_attributes(extra_attrs))
end

  def optional_attributes(nonce: nil, ial: nil, scope: nil)
    { nonce: nonce, ial: ial, scope: scope }
  end
```